onnxruntime/core/optimizer/selectors_actions/actions.h

```diff
@@ -114,13 +117,31 @@ struct ReplaceWithNew : public Action {
                     SavedState& saved_state, bool& graph_modified) const override;
 #endif  // !defined(ORT_MINIMAL_BUILD)
 
- private:
-  // support usage where operator name is determined at runtime from the selected nodes
-  virtual std::string OpType(const NodesToOptimize&) const { return op_; }
+ protected:
+  // contains runtime state that may be used when overriding virtual methods below
+  struct RuntimeState {
+    const Graph& graph;
+    const NodesToOptimize& selected_nodes;
+  };
+
+  // replacement node Op type
+  virtual std::string OpType(const RuntimeState&) const { return op_; }
+
+  // replacement node domain
+  virtual std::string Domain(const RuntimeState&) const { return domain_; }
```

Comment:
Is there a scenario where the derived class would need to a) override these two methods or b) use RuntimeState to get the op_ or domain_? 

I understand it's more consistent, but it's also potentially confusing and slightly more complicated than necessary for OpType and Domain if it's not required.

Reply:
see [https://github.com/microsoft/onnxruntime/blob/e47434ea1281d83e54ea99ef053223aae874ec91/onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_actions.h#L39-L41](https://github.com/microsoft/onnxruntime/blob/e47434ea1281d83e54ea99ef053223aae874ec91/onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_actions.h#L39-L41)

that uses the selected nodes to determine the op type.